runelite-client/src/main/java/net/runelite/client/util/SwingUtil.java

```diff
@@ -282,10 +276,24 @@ public static JButton createSwingButton(
 			button.setComponentPopupMenu(popupMenu);
 		}
 
+		navigationButton.setOnIconChange(() -> setButtonIcon(button, navigationButton.getIcon(), iconSize));
 		navigationButton.setOnSelect(button::doClick);
 		return button;
 	}
 
+	public static void setButtonIcon(
```

Comment:
This can be private. Or moved to SwingUtil

Reply:
It is in SwingUtil.

Reply:
Oh, lol